packages/web/src/onnx.ts

```diff
@@ -1,47 +1,70 @@
 export { createOnnxSession, runOnnxSession };
 
 import ndarray, { NdArray } from 'ndarray';
-import * as ort from 'onnxruntime-web';
+import type ORT from 'onnxruntime-web';
+
+import * as ort_cpu from 'onnxruntime-web';
+// @ts-ignore
+import * as ort_gpu from 'onnxruntime-web/webgpu';
+
 import { loadAsUrl } from './resource';
 import { simd, threads } from './feature-detect';
 import { Config } from './schema';
 
 async function createOnnxSession(model: any, config: Config) {
+  const ort = config.device === 'gpu' ? ort_gpu : ort_cpu;
   const capabilities = {
     simd: await simd(),
     threads: await threads(),
     numThreads: navigator.hardwareConcurrency ?? 4,
     // @ts-ignore
     webgpu: navigator.gpu !== undefined
   };
+
+  const useWebGPU = config.device === 'gpu' && capabilities.webgpu;
+  const useThreads = capabilities.threads;
+  const useSimd = capabilities.simd;
+
+  const executionProviders = [useWebGPU ? 'webgpu' : 'wasm'];
+
   if (config.debug) {
     console.debug('Capabilities:', capabilities);
+    console.debug('Using WebGPU:', useWebGPU);
+    console.debug('Using Threads:', useThreads);
+    console.debug('Using SIMD:', useSimd);
     ort.env.debug = true;
     ort.env.logLevel = 'verbose';
   }
 
   ort.env.wasm.numThreads = capabilities.numThreads;
   ort.env.wasm.simd = capabilities.simd;
-  ort.env.wasm.proxy = config.proxyToWorker;
+  ort.env.wasm.proxy = useWebGPU ? false : config.proxyToWorker;
 
   ort.env.wasm.wasmPaths = {
     'ort-wasm-simd-threaded.wasm':
-      capabilities.simd && capabilities.threads
+      useThreads && useSimd
         ? await loadAsUrl(
-            '/onnxruntime-web/ort-wasm-simd-threaded.wasm',
+            useWebGPU
+              ? '/onnxruntime-web/ort-wasm-simd-threaded.jsep.wasm'
+              : '/onnxruntime-web/ort-wasm-simd-threaded.wasm',
             config
           )
         : undefined,
     'ort-wasm-simd.wasm':
-      capabilities.simd && !capabilities.threads
-        ? await loadAsUrl('/onnxruntime-web/ort-wasm-simd.wasm', config)
+      !useThreads && useSimd
+        ? await loadAsUrl(
+            useWebGPU
+              ? '/onnxruntime-web/ort-wasm-simd.jsep.wasm'
+              : '/onnxruntime-web/ort-wasm-simd.wasm',
+            config
+          )
         : undefined,
     'ort-wasm-threaded.wasm':
-      !capabilities.simd && capabilities.threads
+      !useWebGPU && useThreads && !useSimd
         ? await loadAsUrl('/onnxruntime-web/ort-wasm-threaded.wasm', config)
         : undefined,
     'ort-wasm.wasm':
-      !capabilities.simd && !capabilities.threads
+      !useWebGPU && !useThreads && !useSimd
         ? await loadAsUrl('/onnxruntime-web/ort-wasm.wasm', config)
         : undefined
   };
```

Comment:
I find this code path seems a bit hard to scan, I'd personally like to simplify this logic

Reply:
Feeded your comment to CoPilot and my sidekick wants to change it to this: 
  
  
  const wasmPaths = {
    'ort-wasm-simd-threaded.wasm': useThreads && useSimd,
    'ort-wasm-simd.wasm': !useThreads && useSimd,
    'ort-wasm-threaded.wasm': !useWebGPU && useThreads && !useSimd,
    'ort-wasm.wasm': !useWebGPU && !useThreads && !useSimd
  };

  for (const [key, value] of Object.entries(wasmPaths)) {
    if (value) {
      const wasmPath = useWebGPU && key.includes('simd') ? `/onnxruntime-web/${key.replace('.wasm', '.jsep.wasm')}` : `/onnxruntime-web/${key}`;
      ort.env.wasm.wasmPaths[key] = await loadAsUrl(wasmPath, config);
    }
  }